tests/unit/test_svs_tiered.cpp

```diff
@@ -0,0 +1,2474 @@
+    auto *tiered_index = this->CreateTieredSVSIndex(svs_params, mock_thread_pool);
+    ASSERT_INDEX(tiered_index);
+    auto allocator = tiered_index->getAllocator();
+
+    auto svs = tiered_index->GetBackendIndex();
+    auto flat = tiered_index->GetFlatIndex();
+
+    // Insert 5 vectors to the main index.
+    for (size_t i = 0; i < 5; i++) {
+        GenerateAndAddVector<TEST_DATA_T>(svs, dim, i, i);
+    }
+    // Sanity check. Should choose as SVS.
+    ASSERT_EQ(tiered_index->preferAdHocSearch(5, 5, true), svs->preferAdHocSearch(5, 5, true));
+
+    // Insert 6 vectors to the flat index.
+    for (size_t i = 0; i < 6; i++) {
+        GenerateAndAddVector<TEST_DATA_T>(flat, dim, i, i);
+    }
+    // Sanity check. Should choose as flat as it has more vectors.
+    ASSERT_EQ(tiered_index->preferAdHocSearch(5, 5, true), flat->preferAdHocSearch(5, 5, true));
+
+    // Check for preference of tiered with subset (10) smaller than the tiered index size (11),
+    // but larger than any of the underlying indexes.
+    ASSERT_NO_THROW(tiered_index->preferAdHocSearch(10, 5, false));
+}
+
+TYPED_TEST(SVSTieredIndexTestBasic, runGCAPI) {
+    // Create TieredSVS index instance with a mock queue.
+    size_t dim = 4;
+    size_t threshold = 1024;
+    const size_t n = threshold * 3;
+    SVSParams params = {.type = TypeParam::get_index_type(), .dim = dim, .metric = VecSimMetric_L2};
+    VecSimParams svs_params = CreateParams(params);
+    auto mock_thread_pool = tieredIndexMock();
+
+    auto *tiered_index = this->CreateTieredSVSIndex(svs_params, mock_thread_pool, threshold);
+    ASSERT_INDEX(tiered_index);
+    auto allocator = tiered_index->getAllocator();
+
+    // Insert three block of vectors directly to SVS.
+    std::srand(10); // create pseudo random generator with any arbitrary seed.
+    for (size_t i = 0; i < n; i++) {
+        TEST_DATA_T vector[dim];
+        for (size_t j = 0; j < dim; j++) {
+            vector[j] = std::rand() / (TEST_DATA_T)RAND_MAX;
+        }
+        VecSimIndex_AddVector(tiered_index->GetBackendIndex(), vector, i);
+    }
+
+    ASSERT_EQ(tiered_index->indexSize(), n);
+    ASSERT_EQ(tiered_index->GetBackendIndex()->indexSize(), n);
+
+    // Delete all the vectors and wait for the thread pool to finish running the repair jobs.
+    for (size_t i = 0; i < threshold; i++) {
+        tiered_index->deleteVector(i);
+    }
+    // Launch the BG threads loop that takes jobs from the queue and executes them.
+    mock_thread_pool.init_threads();
+    mock_thread_pool.thread_pool_join();
+
+    ASSERT_EQ(tiered_index->indexSize(), n - threshold);
+    ASSERT_EQ(tiered_index->GetSVSIndex()->indexStorageSize(), n);
+    ASSERT_EQ(mock_thread_pool.jobQ.size(), 0);
+
+    // Run the GC API call, expect that we will clean up the SVS index.
+    VecSimTieredIndex_GC(tiered_index);
+    ASSERT_EQ(tiered_index->indexSize(), n - threshold);
+    ASSERT_EQ(tiered_index->GetSVSIndex()->indexStorageSize(), n - threshold);
```

Comment:
Can we validate the memory consumption of svs before and after running the GC?

Reply:
Done.